libs/SmartStore/SmartStore/Classes/SFSmartStore.m

```diff
@@ -1125,27 +1125,28 @@ - (void)updateTable:(NSString*)tableName values:(NSDictionary*)map entryId:(NSNu
 }
 
 - (NSString*)columnNameForPath:(NSString*)path inSoup:(NSString*)soupName withDb:(FMDatabase*) db {
-    //TODO cache these with soupName:path ? if slow...
     NSString *result = nil;
-    if (nil == path) {
-        return result;
+    NSArray* indexSpecs = [self indicesForSoup:soupName withDb:db];
```

Comment:
Took care of the old TODO on the left by calling indicesForSoup (which does use a cache).